### PR TITLE
[Tests-Only]Improve test docs

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description:
With this PR:
- added how to add tests in docs

Closes https://github.com/owncloud/owncloud-sdk/issues/802

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,24 +75,55 @@ $ yarn build:docs
...
 ### consumer tests
-The pact consumer tests checks owncloud sdk against the pact mock server. In order to run these tests, use this comand
+The pact consumer tests checks owncloud sdk against the pact mock server. In order to run these tests, use the following command:
 
 ```
 $ yarn test-consumer
 ```
 
-If you have pacts from old test run in `tests/pacts` your tests will fail. Make sure to delete that before you run the tests again.
+> If you have pacts from old test run in `tests/pacts` your tests will fail. Make sure to delete that before you run the tests again.
```
Comment (individual-it) [inline README.md:123]: why this change?

File: README.md [hunk 2]
```diff
@@ -75,24 +75,55 @@ $ yarn build:docs
...
 
 ### Provider tests
-The pact provider tests tests the pacts generated from the consumer tests against the real owncloud bakcend server. For this you will need a actual owncloud server running. Then you can run the pact tests with this command.
+The pact provider tests tests the pacts generated from the consumer tests against the real owncloud backend server. For this you will need an actual owncloud server running. Then you can run the pact tests with the following command:
```
Comment (individual-it) [inline README.md:126]:
```suggestion
The pact provider tests the pacts generated from the consumer tests against the real owncloud backend server. For this you will need an actual owncloud server running. Then you can run the pact tests with the following command:
```

**Review by individual-it — CHANGES_REQUESTED** [review]
I would also prefer to split it into 2 PRs, one for docs and one for the rest

File: README.md (hunk 1 above)
Comment (kiranparajuli589) [reply README.md:123]: this is kind of a note rather than how to run.

**Review by individual-it — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -75,24 +75,55 @@ $ yarn build:docs
...
+- interactions that work on both oc10 & ocis
+- interactions that are "pending" on oc10 but should work on ocis
+- interactions that are "pending" on ocis but should work on oc10
+- interactions that are "pending" on both ocis & oc10
+
+The CI is not expected to fail for the interactions that are "pending" but it's expected to fail for those interactions that were already verified and started failing. Pact.io has a system to handle such a scenario called [pending pacts](https://docs.pact.io/pact_broker/advanced_topics/pending_pacts/ 'pending pacts'). This feature allows changed contracts to be verified without failing the provider's build.
+
+Four different pacts for the different buckets of interactions are created when running the provider tests. Pacts that are allowed to fail are marked as `pending` else not.
+
+In a consumer test, a new provider is created using the function `createProvider`. It is needed to be called with two parameters, depending upon which provider-version the test is still pending.
```
Comment (saw-jan) [inline README.md:92]:
```suggestion
In a consumer test, a new mock provider is created using the function `createProvider`. It takes two parameters: `pendingOnOc10` and `pendingOnOcis` in order. Each parameter can have value (true or false) depending upon which provider-version the test is still pending.
```

**Review by saw-jan — COMMENTED** [review]
LGTM
